### DISCUSSION TITLE
How to use Sonoff RF (on Sonoff 4CH Pro RF R3)

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#15365] [Q&A] [answered]
Need some support.
I flashed the Sonoff 4CH Pro RF R3 with tasmota(~~bluetooth~~ sensors _stupid mistake_) .
Did this [connection](https://user-images.githubusercontent.com/3578202/139196247-4f67058f-3f2f-4d8e-81e7-f08dfc9595bb.jpg)
[Sonoff 4CH Pro](https://tasmota.github.io/docs/devices/Sonoff-4CH-Pro/#sonoff-4ch-pro-r3)

I can't find the pairing mode for rf. Tried double push on buttons and the 7 seconds push (or twice 7 seconds pushes).
Also i see no alive (Output) from RF modul in console.
Since i configured GPIO 2 as RFrecv i could set RFProtocol . But i did not find any Informations about the RFProtocol paramters.

Maybe, someone could help me to get startet. i would like to use a cheap chinese rf remote.

thanks a lot in advance.

**Comment by barbudor** [comment]
~~tasmota32-bluetooth doesn't include USE_RC_SWITCH which is required for this feature. Only tasmota32 does.
As you said you were able to configure GPIO to RFRecv, I would assume that you did compile your own firmware.~~
Sonof 4CH Pro R3 is using a ESP8266 so you probably flashed "tasmota-sensors" so I don't get your point with "bluetooth"

There is no pairing but you should get a "RF Received" message in the console when a RF code is received, as per the documentation.

What is the other device sending the RF message ? A remote ? 
Are you sure its protocol is supported by the RCSwitch library variant used by Tasmota ?

**Comment by ChrisVonFrosta** [comment, marked as answer]
Thx for reaction!
Bluetooth was stupid. i ment sensors.

I had some simple remote wall switches which works "ok" with the sonoff stock firmware before.
I have either no knowledgement about its protocol, nor about the different supported RF protocols from Tasmota.
Can't find documentation about this.
Only that [site](https://tasmota.github.io/docs/RF-Protocol/#receiving-rf-codes)

**Reply by barbudor** [reply]
The list of supported protocols are unfortunately in the source code and hasn't been copied into the Doc: 
https://github.com/arendst/Tasmota/blob/b980e93a8616cbff114a2c284eab124f97f605cc/lib/lib_rf/rc-switch/src/RCSwitch.cpp#L116-L154

If your remote is not supported by the RC-Switch library then to add it requires the tools such as a logic analyser to analyse your remote  signal and create a descriptor for it.
